File: domain/src/main/java/be/insaneprogramming/cleanarch/entity/TenantFactory.java
package be.insaneprogramming.cleanarch.entity;

import java.util.UUID;

public class TenantFactory {
	public Tenant createTenant(String id, String name) {
		return new Tenant(id, name);
	}

	public Tenant createTenant(String name) {
		return createTenant(UUID.randomUUID().toString(), name);
	}
}

File: domain/src/main/kotlin/be/insaneprogramming/cleanarch/entity/TenantFactory.kt
package be.insaneprogramming.cleanarch.entity

import java.util.*

class TenantFactory {
    fun createTenant(id: TenantId, name: String): Tenant {
        return Tenant(id, name)
    }

    fun createTenant(name: String): Tenant {
        return createTenant(TenantId(UUID.randomUUID().toString()), name)
    }
}
